sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -229,30 +229,48 @@ export function getBooleanOrUndefined(value: any): boolean | undefined {
 }
 
 /**
- *  @ignore
- * Helper utility to check for and return JSON like object,
- * or undefined if not passed in. Treats empty string as `undefined`.
+ * @ignore
+ * Helper utility to check for and return valid JSON like object i.e.,
+ * a single JSON like object or array of JSON like objects.
  * @param value
  */
-export function getJSONOrUndefined(value: any): any | undefined {
-  if (value == undefined || (typeof value === "string" && value.trim() === "")) {
+export function getJSObjectOrUndefined(value: any): any | undefined {
+  if (value == undefined) {
     return undefined;
   }
 
-  const formattedStringValue = JSON.stringify(value);
-  if (!formattedStringValue.startsWith("{") && !formattedStringValue.startsWith("[")) {
-    throw new TypeError(
-      `${formattedStringValue} expected to be in proper JSON format, or undefined`
-    );
+  let isValidJSObject = true;
+
+  if (Array.isArray(value)) {
+    for (let i = 0; i < value.length; i++) {
+      if (!isJSONLikeObject(value[i])) {
+        isValidJSObject = false;
+        break;
+      }
+    }
+  } else {
+    if (!isJSONLikeObject(value)) {
+      isValidJSObject = false;
+    }
   }
 
-  try {
-    return JSON.parse(formattedStringValue);
-  } catch (err) {
-    throw new TypeError(
-      `Error parsing JSON - ${JSON.stringify(value, undefined, 2)} : ${err.message} `
-    );
+  if (!isValidJSObject) {
+    throw new TypeError(`${value} expected to be in proper JSON format, or undefined`);
   }
+
+  return value;
+}
+
+/**
+ * Returns `true` if given input is a JSON like object.
+ * @param value
+ */
+function isJSONLikeObject(value: any): boolean {
```

Comment:
JSON.stringify should not be used in this way I think. It's very expensive for, if I understand correctly, returning false for String and Number objects.

Given that String and Number are simply not used in practice, I'm ok not handling them. If it seems super important, then I think this function would be better implemented like:

```
return typeof value === 'object' && !value instanceof Number && !value instanceof String;
```

Reply:
Updated